zebrad/src/components/sync/tests/timing.rs

```diff
@@ -78,6 +79,19 @@ fn ensure_timeouts_consistent() {
         "a syncer tip crawl should complete before most new blocks"
     );
 
+    // This is a compromise between two failure modes:
+    // - some peers have the inventory, but they weren't ready last time we checked,
+    //   so we want to retry soon
+    // - all peers are missing the inventory, so we want to wait for a while before retrying
+    assert!(
+        INVENTORY_ROTATION_INTERVAL < SYNC_RESTART_DELAY,
+        "we should expire some inventory every time the syncer resets"
+    );
+    assert!(
+        2 * INVENTORY_ROTATION_INTERVAL > SYNC_RESTART_DELAY,
+        "we should expire all inventory after the syncer has reset once or twice"
```

Comment:
Question: shouldn't this assertion be something like:

```suggestion
        2 * INVENTORY_ROTATION_INTERVAL < 3 * SYNC_RESTART_DELAY,
        "we should expire all inventory after the syncer has reset once or twice"
```

Reply:
I think the equation is correct, but the wording is unclear.

(What matters is that there is at least one retry with some of the previous inventory registered, so that the syncer gets a chance to try better peers. There could be two retries with some of the previous inventory, but the second retry is not required.)

Clarified in 3589d5c33